Automattic/wc-calypso-bridge
# Deploy Automation: Automate doing a release build.

### Changes proposed in this Pull Request:

This adds an initial command to checkout the latest `master` branch and do an `npm run build`. There is some rudimentary checking to ensure the build completed properly.

Closes #1100
Related to https://github.com/Automattic/wc-calypso-bridge/issues/1096






### How to test the changes in this Pull Request:




1. Clone this branch.
2. In the `./scripts` directory, run `yarn` to manually install the script dependencies.

There are a number of different scenarios to check:

### Basic canceling of the script
1. In the project root, run `yarn create-release`.
1. At the initial confirmation prompt, type 'N' and verify the script stops running.
![image](https://user-images.githubusercontent.com/917632/235184121-df19b2e3-2324-4f64-8320-8c4f5db6c344.png)

### Incorrect node version
1. Switch to a NodeJS version < v16.
1. In the project root, run `yarn create-release`.
1. You should get an `ERROR: Your version of NodeJS is not correct. Please install NodeJS v16.` error and the script should stop.

### Unclean working copy
1. Now make any sort of change in the project directory (a simple `touch t` to create a new file is fine).
1. In the project root, run `yarn create-release` and type 'y' to continue.
1. You should get a `ERROR: Working copy is not clean. Please commit or stash your changes before releasing.` error and the script should stop.

### Dev build is present
1. In the project root, run `npm i && npm start`. You can CTRL-C to cancel the `npm start` once it's done compiling.
1. In the project root, run `yarn create-release` and type 'y' to continue.
1. You should get an `ERROR: You may have a dev build. Please make sure you aren't running 'npm start'` error and the script should stop.

### Successful build
1. In the project root, run `rm -rf ./build` to clear out the dev build files.
1. In the project root, run `yarn create-release` and type 'y' to continue.
1. The script should run until you see a `SUCCESS: Release build complete` message.
1. Verify that the project root has a `build/` directory without any `.map` or empty files.
1. Verify that the `node_modules/` directory in the project root has been removed.
1. Verify that you are on the `master` branch.



### Other information:




### FOR PR REVIEWER ONLY:
